### PR TITLE
Issue #7605: Update examples for SingleLineJavadoc check

### PR DESCRIPTION
**Pull request by ritvij14**
Website generated:

Screenshot (11)
Screenshot (12)
Screenshot (13)

1.) The output of default example:
config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
   <module name="SingleLineJavadoc"/>
  </module>
</module>
```

Test.java
```
public class Test {
  /**@return integer value*/
  public int sum(int a, int b) {
    return a + b;
  }
}
```
```
C:\Users\hp\Desktop>java -jar ../Downloads/checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\hp\Desktop\Test.java:2: Single-line Javadoc comment should be multi-line. [SingleLineJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```

2.) Output of non-default example for violateExecutionOnNonTightHtml
config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
   <module name="SingleLineJavadoc">
     <property name="violateExecutionOnNonTightHtml" value="true"/>
   </module>
  </module>
</module>
```

Test.java
```
public class Test {
  /**
   *@return integer value
   *<p> First
   */
  public int sum(int a, int b) {
    return a + b;
  }
} //Prints violation for unclosed HTML tag.
```
```
C:\Users\hp\Desktop>java -jar ../Downloads/checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\hp\Desktop\Test.java:4: Unclosed HTML tag found: p [SingleLineJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```

3.) Output for non-default example for ignoredTags and ignoredInlineTags
config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
   <module name="SingleLineJavadoc">
     <property name="ignoredTags" value="@inheritDoc, @see"/>
     <property name="ignoreInlineTags" value="false"/>
   </module>
  </module>
</module>
```
Test.java
```
public class Test {
  /** @see #sum(int a, int b) */
  public int sum(int a, int b) {
    return a + b;
  }
}
```
```
C:\Users\hp\Desktop>java -jar ../Downloads/checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```
Test.java
```
public class Test {
  /** Testing class for {@link Example} */
  public int sum(int a, int b) {
    return a + b;
  }
}
```
```
C:\Users\hp\Desktop>java -jar ../Downloads/checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\hp\Desktop\Test.java:2: Single-line Javadoc comment should be multi-line. [SingleLineJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```
```

### CODE REVIEW

**Comment by romani** [comment]
@ritvij14 , CI is failed - please look at errors and resolve all of them.

your commit message does not follow required template. 
please read https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SingleLineJavadocCheck.java [hunk 1]
```diff
@@ -58,6 +58,45 @@
...
  * &lt;module name=&quot;SingleLineJavadoc&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   /** &#64;return integer value *&#47;
+ *   public int sum(int a, int b) {
+ *     return a + b;
+ *   }
+ * } //Violation. If Javadoc contains at least one at-clause, it should be formatted in multiple lines.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SingleLineJavadocCheck.java]: please place `// violation` comment on line were violation is placed by CLI tool.
it should be somethig like:
`/** &#64;return violation, when at least one at-clause, it should be formatted in multiple lines. *&#47;`

**Comment by ritvij14** [comment]
Sir, I had some doubts about the fixes to the build fails. I can't understand what to do about the team city build.

Also in the AppVeyor build, I see that there are some errors with the files that I edited, I will fix them. But it also shows some errors with a file java.g in the grammar folder. How should I fix that? It was also shown in red color on the site.

And Travis says that my PR is based on a master that is 14 commits old. I don't understand this part because I just forked from it a couple of days ago.

Also, sir mvn clean verify gave me one error and I found it in the surefire-reports folder. It's this one. How should I fix it?

[com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.txt](https://github.com/checkstyle/checkstyle/files/4356082/com.puppycrawl.tools.checkstyle.internal.XdocsJavaDocsTest.txt)

**Comment by bmhm** [comment]
> And Travis says that my PR is based on a master that is 14 commits old. I don't understand this part because I just forked from it a couple of days ago.


Probably there were at least 14 changes since then. Just rebase your fork on the current master and it will succeed. You can find two links on gitter.

**Comment by ritvij14** [comment]
@bmhm sir, I did what was asked in the link on Gitter. And I also did the changes that were asked by Roman sir. But I don't see my 2nd commit here. Shouldn't it come after I push it again as was said in the Medium blog post?

**Comment by bmhm** [comment]
> Shouldn't it come after I push it again as was said in the Medium blog post?

Hello @ritvij14 - it should be visible after doing a `git commit` and then `git push`. Look for the output message of both commands and no worries to try again, they will do no harm if you executed them earlier. This is called "idempotent" (meaning: they will enforce the same state no matter how often you executed them).

If you did this, just execute `git status`. Make sure you are on the branch  `single-line-javadoc`.
Also type `git remove -v` and look if `origin` is set to your repository, which should be `git@github.com:ritvij14/checkstyle.git`.
I would also guess that you also have another remove which is called `upstream` and points to `git@github.com:checkstyle/checkstyle.git`. If it is not called upstream, this is no problem. It is just a name, just a label. The mechanisms will be the same.

I know in gitter there was a link to medium.com. I haven't read it myself, but you might want to check if the instructions are similar to this page: https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork

This is the manual I usually refer to. I would expect the contents are similar to the description on medium.com.

Let me know if you have any findings.

**Comment by ritvij14** [comment]
@bmhm sir, the Medium post had different commands. Yes, I had the upstream and origin all set. From the link, you shared with me, when I did the `git merge upstream/master`, it started updating my master. And now my single-line-javadoc branch shows nothing to commit, while my master shows

```
On branch master
Your branch is ahead of 'origin/master' by 33 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean
```

**Comment by bmhm** [comment]
> @bmhm sir, the Medium post had different commands. Yes, I had the upstream and origin all set. From the link, you shared with me, when I did the `git merge upstream/master`, it started updating my master. And now my single-line-javadoc branch shows nothing to commit, while my master shows
> 
> ```
> On branch master
> Your branch is ahead of 'origin/master' by 33 commits.
>   (use "git push" to publish your local commits)
> 
> nothing to commit, working tree clean
> ```

Good, so now your master is up to date. You could push it already to github.

Now, for your branch you should try to rebase it to the current master. **This is a good point to backup your local checkstyle in case it does not work easily**. 

If you have your backup, switch to your branch `single-line-javadoc` using `git checkout single-line-javadoc` and then execute `git rebase master`.
Git will then **try** (!) to move the changes you made to the new starting point. This means it will try to move all your changes to a newer commit from master, as if you have forked the repository right now.
If it succeeds, `git status` should output "1 commit ahead" or so. Good reading with some figures are these tutorials:

* https://git-scm.com/book/en/v2/Git-Branching-Rebasing
* https://www.atlassian.com/git/tutorials/merging-vs-rebasing (chapter "the rebase option").

In case did not work, just restore your working copy for now.

**Comment by ritvij14** [comment]
@bmhm sir, I did that. This is the output I am getting.
```
$ git rebase master
Current branch single-line-javadoc is up to date.
```
```
$ git status
On branch single-line-javadoc
nothing to commit, working tree clean
```
Also, pushing it again says that everything is up-to-date.

**Comment by bmhm** [comment]
Hello @ritvij14 
The master branch will not affect this pull request.

Do you have changes on the master branch? You can check this via `git log`.
For example:
```
$ git log --pretty=oneline | head
bcb736e45231e64a9c9b59269e072ffe9892f2c2 Issue #7621: Update doc for GenericWhitespace
12c6e69fe7de3d462f5ca117dc4a6f0ba51a1d7c Issue #7591: Update doc for TodoComment
```

If you see your commits on your master branch, you commited your changes accidentily on the master branch instead of your feature branch.
If you see your commits on your feature branch, please paste the output of your `git status`. Please use three backticks to have code formatting.

**Comment by romani** [comment]
@ritvij14 ,  if you donot know how to fix this, please backup all your changes, drop forks and your local clone and do all from scratch - https://checkstyle.org/beginning_development.html

**Comment by ritvij14** [comment]
@bmhm sir, I checked the `git log` of both my branches. My commits are being shown only in the branch `single-line-javadoc` that I created. This is the output of `git status`.

```
hp@DESKTOP-BFAE7Q7 MINGW64 ~/Desktop/Github/checkstyle (single-line-javadoc)
$ git status
On branch single-line-javadoc
nothing to commit, working tree clean
```

**Comment by bmhm** [comment]
Someone in gitter chat pointed out you have  too many commits on your branch. You might need to squash them first.

**Comment by ritvij14** [comment]
@bmhm sir, I will use Roman sir's advice and drop my current fork and clone. I don't know how but suddenly while squashing my commits, it suddenly started showing 21 commits.

**Pull request closed without merging**